src/main/java/com/tabgok/entity/Machine.java

```diff
@@ -0,0 +1,54 @@
+package com.tabgok.entity;
+
+import com.tabgok.data.StringData;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * The machine is a model of a machine.  Importantly, the machine is the top-level
+ * model for a machine, meaning it contains within it either references to, or values of,
+ * everything within a machine from Hardware to Filesystems to applications.
+ */
+public class Machine extends Entity {
+    private StringData MachineID;
```

Comment:
a mutable ID on an object "smells" like we should rethink the object model here. it can force us to worry about all sorts of shit later on that we won't want to.

do you have a reason for having it be mutable?